### ISSUE TITLE
[BOOM-5944] Split order - order confirmation

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#9594]
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5944).

- _**Reporter:**_ Centoasa
- _**Created at:**_ Tue, 10 Jul 2018 15:24:27 +0200

<p>My customer have bought some products splitted into 2 orders because there are 2 different carriers.</p>

<p>The real sum of all products bought is <b>€967.93,</b> instead prestashop 1.7.4 show me *<b> that the customer should have paid me the sum of the 2 different orders, so  </b><b>  </b>€1,196.80* <b>(wrong!)</b></p>

<p>I hope I'm clear</p>

---
### Steps to reproduce
If we add in cart:

1. Product 1 (which have an available carrier S1)
2. Product 2 (which have an available carrier S2)

=> In the order confirmation, only the product 1 will be displayed in the section "ORDER ITEMS"</li>
If we add in cart:

1. Product 2 (which have an available carrier S2)
2. Product 1 (which have an available carrier S1)

=> In the order confirmation, only the product 2 will be displayed in the section "ORDER ITEMS"

**Screenshots**

https://drive.google.com/file/d/1mKDDAGUyH5prlaNYuuVIF3nSSbkXwMFg/view

**Addional infos**
- PS version: 1.7.7.1
- PHP version: 7.2

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171307).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Tue, 10 Jul 2018 15:43:48 +0200

<p>Hi Remigio,</p>

<p>Thanks for your report.<br/>
I attached a video record.<br/>
 <span class="nobr"><a href="http://forge.prestashop.com/secure/attachment/69317/69317_1740_2+%281%29.webm" title="1740_2 (1).webm attached to BOOM-5944">1740_2 (1).webm<sup><img class="rendericon" src="http://forge.prestashop.com/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span> <br/>
Thanks to check and feedback.</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171309).

- _**Author:**_ centoasa
- _**Created at:**_ Tue, 10 Jul 2018 15:49:22 +0200

<p>Thank you <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=khouloud.belguith" class="user-hover" rel="khouloud.belguith">Khouloud BELGUITH</a> for your video.</p>

<p>I was attached the wring image: now is the right image!</p>

<p>Please ceck it</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171329).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Tue, 10 Jul 2018 16:59:45 +0200

<p>Hi Remigio,</p>

<p> I did not manage to reproduce the issue with the last version 1.7.4.0.<br/>
 <span class="image-wrap" style=""><a id="69324_thumb" href="http://forge.prestashop.com/secure/attachment/69324/69324_screenshot-1.png" title="screenshot-1.png" file-preview-type="image" file-preview-id="69324" file-preview-title="screenshot-1.png"><img src="http://forge.prestashop.com/secure/thumbnail/69324/_thumb_69324.png" style="border: 0px solid black" /></a></span> <br/>
This issue occurs only when you use the payment method Paypal?</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171331).

- _**Author:**_ centoasa
- _**Created at:**_ Tue, 10 Jul 2018 17:09:21 +0200

<p>This is the first time this issue happens, and this time with Paypal.</p>

<p>I don't know with other payments way</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171336).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Tue, 10 Jul 2018 17:39:33 +0200

<p>Hi Remigio,</p>

<p>Could you please check your issue with another payment method for example "payment by check" and feedback.</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171340).

- _**Author:**_ centoasa
- _**Created at:**_ Tue, 10 Jul 2018 17:55:26 +0200

<p>just cecked with bank wire payment: no alert, only split order into 2, with 2 order confirmations and 2 bank wire request.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171341).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Tue, 10 Jul 2018 18:00:33 +0200

<p>Hi Remigio,</p>

<p>The Paypal module isn't developed by PrestaShop, but by one of our partner.<br/>
You can contact them here: <a href="https://support.202-ecommerce.com/fr/support/home" class="external-link" rel="nofollow">https://support.202-ecommerce.com/fr/support/home</a><br/>
Thanks to check the issue with module author and feedback.</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171342).

- _**Author:**_ centoasa
- _**Created at:**_ Tue, 10 Jul 2018 18:01:10 +0200

<p>and why in the customer order confirmation template it refer to one of 2 splitted orders? <span class="image-wrap" style=""><img src="http://forge.prestashop.com/secure/attachment/69325/69325_2018-07-10+%284%29.png" style="border: 0px solid black" /></span></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171359).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Wed, 11 Jul 2018 10:28:42 +0200

<p>Hi Remigio,</p>

<p>Thanks for your feedback.<br/>
I manage to reproduce this issue with the last version 1.7.4.0.</p>
<ol>
	<li>If we add in cart:<br/>
Product 1 (which have an available carrier S1)<br/>
Product 2 (which have an available carrier S2)<br/>
=> In the order confirmation, only the product 1 will be displayed in the section "ORDER ITEMS"</li>
	<li>If we add in cart:<br/>
Product 2 (which have an available carrier S2)<br/>
Product 1 (which have an available carrier S1)<br/>
=> In the order confirmation, only the product 2 will be displayed in the section "ORDER ITEMS"</li>
</ol>


<p>We will see how to fix it.</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171410).

- _**Author:**_ marion_francois
- _**Created at:**_ Wed, 11 Jul 2018 14:32:45 +0200

<p>Hello <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=khouloud.belguith" class="user-hover" rel="khouloud.belguith">Khouloud BELGUITH</a></p>

<p>The issue is just the product displayed in the order confirmation ?</p>

<p>There is no issue with the sum as indicated in the description ?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5944#comment-171412).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Wed, 11 Jul 2018 14:41:23 +0200

<p>Hi <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=marion_francois" class="user-hover" rel="marion_francois">Marion F</a>,</p>

<p>The issue is about "page-order-confirmation card".<br/>
Only one product "Product 1" displayed, with Shipping "S1" => the first order created in details is displayed.<br/>
But the second order related to the first one, (Product P2 with Shipping "S2") is not displayed.</p>

<p>Best regards, Khouloud</p>

**Comment by centoasa** [comment]
same issue again.
customer have purchased 2 products: 
product A price 73 shipped by couries A (shipping cost 8,50)
product B price 151,53 shipped by courier B (shipping cost 12)
Scenario: customer closed own order payed by Paypal; email confirmation was right:
one email contain product 73 + shipping 8.50 = totale 81.50
other email contain product 151.50 +12= total 163.50
Problem is on prestashop backoffice, 2 split orders show me:
73+8.50=245.03 (?????) => https://1ce.org/1#Sy6UsGiIX
other 151.53+12=163.53 ===> https://1ce.org/1#Hy9Tjzi8m

Note: custoemr have purchased is on group "dealer" that show price net.

**Comment by romain-le-mercier** [comment]
Don't know if it's the same issue but I also have order split in two.
Product are restricted by the same carrier, on both the order the carrier is the same.
The difference in my case is that there is always a discount that offer a product and free shipping.
I have his problem on 4 order, 2 paypal and two Debit card.

**Comment by COCOPOT** [comment]
Hi, I got the same error. It doesn´t matter the payment method. Does anyone find a solution for this? thank you. Also got a PrestashopException when trying to adjust amount with adding or editing a discount in order. See attached pictures.
![captura de pantalla 2019-02-12 a las 23 30 49](https://user-images.githubusercontent.com/47578734/52672493-5601e580-2f1e-11e9-8b5a-bfec410f6ab1.jpg)
![captura de pantalla 2019-02-12 a las 23 02 36](https://user-images.githubusercontent.com/47578734/52672494-5601e580-2f1e-11e9-8458-69bd9b365f96.jpg)
![captura de pantalla 2019-02-12 a las 22 37 02](https://user-images.githubusercontent.com/47578734/52672496-569a7c00-2f1e-11e9-9526-c49bd8fc4324.jpg)
![captura de pantalla 2019-02-12 a las 22 36 33](https://user-images.githubusercontent.com/47578734/52672497-569a7c00-2f1e-11e9-8a81-90c9ae11937c.jpg)
![captura de pantalla 2019-02-12 a las 22 54 12](https://user-images.githubusercontent.com/47578734/52672500-569a7c00-2f1e-11e9-81a8-77c89b461b23.jpg)

**Comment by khouloudbelguith** [comment]
Hi @COCOPOT,

About your last screenshot.
We are aware of this issue, it is added to our bug roadmap.
You need to follow this issue: #11059.
Thanks!

**Comment by sergiogarciadisenoweb** [comment]
hi @COCOPOT , can fix your problem? i have same issue with 1.7.5.0 version. in some orders the order split in 2 orders with 1 sec diference. the orders have same reff. not all orders, aprox 1%

**Comment by khouloudbelguith** [comment]
Hi @sergiogarciadisenoweb,

Sorry not yet.
Thanks for your understanding and your patience.

**Comment by COCOPOT** [comment]
Hi, @sergiogarciadisenoweb ,
The problem is not solved but I am sure at 99% that it´s a problem with Paypal module. I am dealing with this error every day, I have to modify orders´amount in database to fix quantities, and the error is always with orders made with Paypal module.
I wrote to Paypal official module developers to explain the problem and gave them access to our BO store so they can see it but we have no answer.

**Comment by COCOPOT** [comment]
Hi!
I am glad to inform that people from 202 ecommerce, developers of Official PayPal module, solved the problem in our store. Now Order Amount is right when products are mapped with the different shipping method.
I have invited them to parcicipate here and to share the sollution wich I suppouse will be availble in next module version.
Thank you all,
Best regards,

**Comment by rrataj** [comment]
Hello! 
I'm also facing the problem with PayPal module and splitting orders with wrong amount. Can you share your solution? The new version of PayPal module is not yet released.

Best Regards

**Comment by COCOPOT** [comment]
Hi, you have to write to module developers. They will solve it. I asked them to share here the solution but they told me that they only answer to customers via prestashop addons support.

Regards,

**Comment by marionf** [comment]
In the order conf in FO, obly one product is displayed instead of 2

![capture d'écran_1522](https://user-images.githubusercontent.com/13449658/57679876-a4617600-762c-11e9-895c-4c20c2e46f61.png)


In BO, I well have 2 orders created and amounts are ok

![capture d'écran_1523](https://user-images.githubusercontent.com/13449658/57679940-c955e900-762c-11e9-8de1-3c3c755fb1f9.png)

![capture d'écran_1524](https://user-images.githubusercontent.com/13449658/57679944-cbb84300-762c-11e9-964f-17cbde1baedc.png)

**Comment by guigplt** [comment]
Hi,
I have the same problem of order splitted in two.
It happened 13 times in 3 months, so it doesn't happen often (about 0,4% of orders).

I don't see any common point between orders : it happened with payment methods PayPal or Stripe, different shipping methods, and different products.

Regards.

**Comment by guigplt** [comment]
Hi,
Is there a fix for this bug ?

Regards.

**Comment by COCOPOT** [comment]
I got this fixed. I asked oficial developers of Paypal module to fix it and they did the job. I only got this issue with Paypal module.

Regards,

**Comment by guigplt** [comment]
Can you tell me when it has been fixed ?
Because I have module version 4.4.5 on Prestashop 1.7.5.1.

And I have this bug on other payment methods :

![image](https://user-images.githubusercontent.com/44002621/65311810-7ae3c880-db91-11e9-9f9b-5492834ae2b6.png)

Regards.

**Comment by rrataj** [comment]
@guigplt What `Round mode` and `Round type` do you have in Shop parameters -> General ? Please try `Round up away from zero, when it is half way there (recommended)` and `Round on each item` settings.

**Comment by guigplt** [comment]
Hi,
My configuration is :
![image](https://user-images.githubusercontent.com/44002621/65312734-6ef90600-db93-11e9-8443-4647beb1fcad.png)

I will try Round on each item.
What's the difference ?

Regards.

**Comment by rrataj** [comment]
@guigplt This is the first thing you need to check before we start checking other things. Often payment modules and prestashop does not calculate rounding in the same way, and this could lead to these kind of splitting issues as well. It is also written in official PayPal documentation :) Hope it will help you.

**Comment by guigplt** [comment]
Ok thank you for your help !

**Comment by guigplt** [comment]
I configured as recommanded (Round up away from zero, when it is half way there (recommended) and Round on each item), but the issue is still here.
It happened two times since I changed the settings.
Regards.

**Comment by tristanbenier** [comment]
Hi there,

I am on Prestashop 1.7.6.1 and the problem occurs randomly.
@COCOPOT I have only one payment module, which is not Paypal.

I also have the "recommended" configuration for price rounding calculation.

```
+---------------------+-----------------------------+-------+
| date_add            | payment                     | count |
+---------------------+-----------------------------+-------+
| 2020-02-05 11:22:00 | Paiement par carte bancaire |     2 |
| 2020-01-31 16:11:23 | Paiement par carte bancaire |     2 |
| 2019-11-29 22:18:23 | Paiement par carte bancaire |     2 |
| 2019-11-28 13:19:48 | Paiement par carte bancaire |     2 |
| 2019-11-07 21:49:23 | Paiement par carte bancaire |     2 |
| 2019-06-11 10:35:35 | Paiement par carte bancaire |     2 |
| 2019-05-17 12:22:43 | Paiement par carte bancaire |     2 |
| 2019-05-13 18:15:06 | Paiement par carte bancaire |     2 |
| 2019-05-13 18:15:05 | Paiement par carte bancaire |     2 |
| 2019-05-11 20:45:44 | Paiement par carte bancaire |     2 |
| 2019-05-11 20:45:43 | Paiement par carte bancaire |     2 |
| 2019-04-30 23:22:40 | Paiement par carte bancaire |     2 |
| 2019-03-10 18:38:23 | Paiement par carte bancaire |     2 |
| 2019-01-30 21:32:48 | Paiement par carte bancaire |     2 |
| 2019-01-30 00:14:56 | Paiement par carte bancaire |     2 |
| 2019-01-30 00:14:55 | Paiement par carte bancaire |     2 |
| 2019-01-28 22:49:01 | Paiement par carte bancaire |     2 |
| 2019-01-07 22:01:41 | Paiement par carte bancaire |     2 |
| 2018-12-18 20:26:32 | Paiement par carte bancaire |     2 |
| 2018-12-18 20:26:31 | Paiement par carte bancaire |     2 |
+---------------------+-----------------------------+-------+
```

**Comment by Tarraga** [comment]
Hello, I'm using the latest stable version of PrestaShop and it still has the order confirmation bug when split order is applied. Only 1 product shows. Does it exist any solution for this?

**Comment by aestuar** [comment]
Same here, need also a solution.

**Comment by rrataj** [comment]
In general AFAIK there are couple of bug related to splitting orders in PS (We are using 1.7.5.1 and multi warehouse config). There are solutions for most of them, but it's custom code implemented by me and I can't guarantee it's working 100% reliably, so I won't share it at this stage. I'm also not sure anymore what this thread is really about, because I saw multiple bugs mixed up together here.

For order confirmation page: the idea is to show products and totals from Cart object associated to orders, not Order object, because Order object used for presenting data on order confirmation page is always only one of all auto splitted orders (first or last, can't remember which one exactly). 

If you would like to see some code examples let me know.

**Comment by ComonSoft** [comment]
Same issue. Random orders are splitted.

1/ no specific carrier on any products
2/ happends with Paypal, Check, Visa...
3/ cart is also spillted with no carrier

PS 1.7.4.2 PHP 7.2

**Comment by guigplt** [comment]
Hello,
Is there any news about this issue ?
Still happening randomly...

**Comment by Sameye** [comment]
Hi,
I've just experienced the same issue. The order was split in 2 and I've checked the products carrier, all was ok.
The customer even paid twice the carrier and I have a module to log in to customer account, I've made the exact same order and all is ok (only one carrier and carrier fees are added only once).
I've made several Prestashop website but it's the first time I have this issue, is there something new about it?
Thanks.

**Comment by COCOPOT** [comment]
Hi,
Finally we managed to solve this problem, but in a very expensive way, by using Webkul "Marketplace" module and Webkul "Order split" addon for Marketplace, and a little customization by a programmer. Both, module and addon, are very expensive, but we needed a solution for this. Now when we have more than one carrier involved in an order, we configure the modules to deal with them as if there where two different sellers in the store. This is very cool because customer can see from wich warehouse will be sent each product before finish order. The order is splitted in 2, 3 or more parts, depending on all carriers involved and the problem with paiment is solved.

**If you are interested to check this in live, add this products to cart and go to checkout:**
**Product (1): Sent from external warehouse and with a free shipping carrier:**
https://www.cocopot.es/cultivo-setas/1474-bala-setas-de-pleurotus-ostreatus-17kg-germinada-0643415976845.html
**Product (2): Sent from our warehouse, with two shipping options (normal carrier or preferent) customer can select carrier:**
https://www.cocopot.es/germinador-semillas/550-semillero-germinador-grande-58x40x225cm-5031670003034.html
**Product (3): Sent from another external warehouse, with another free shipping carrier:**
https://www.cocopot.es/huerto-urbano/5020-cajonera-de-cultivo-hortalia-160x80cm-altura-30cm-8436562750152.html

Also, I have to say that after installing modules, it didn´t worked at all. Webkul support was very slow and not very effective. We also bought his OnePageCheckout module but it was so ugly and not very good so we decided to work with a programmer to modify our actual OnePageCheckout module to make it compatible with webkul modules. And we are very happy with the result. It´s working very well.

It is not a cheap solution at all, but we need it working. Hope this can help you,
Regards,

**Comment by seederp2p** [comment]
Just to confirm, this issue is still present on 1.7.7.2 correct?

I faced this problem on 1.7.7.2 with vivawallet payment, but I don't see the relation with payment method because:

If we add in cart:

Product 1 (which has an available carrier Carrier1)
Product 2 (which has an available carrier Carrier2, Carrier3, Carrier 4)
=> In the order checkout STEP 3 it only shows 1 carrier (example: Carrier2) but cart still has 2 products.

If you invert the order where you insert the product in cart like:

Product 2 (which has an available carrier Carrier2, Carrier3, Carrier 4)
Product 1 (which has an available carrier Carrier1)
=> In the order checkout STEP 3 it only shows 1 carrier without name but cart still has 2 products.

It allows you to make the order and at the backoffice level it splits the order.

This is very severe because if you don't allow S1 for a specific country, you still end up the order and the customer makes the payment.

**Comment by aestuar** [comment]
> Just to confirm, this issue is still present on 1.7.7.2 correct?
> 
> I faced this problem on 1.7.7.2 with vivawallet payment, but I don't see the relation with payment method because:
> 
> If we add in cart:
> 
> Product 1 (which has an available carrier Carrier1)
> Product 2 (which has an available carrier Carrier2, Carrier3, Carrier 4)
> => In the order checkout STEP 3 it only shows 1 carrier (example: Carrier2) but cart still has 2 products.
> 
> If you invert the order where you insert the product in cart like:
> 
> Product 2 (which has an available carrier Carrier2, Carrier3, Carrier 4)
> Product 1 (which has an available carrier Carrier1)
> => In the order checkout STEP 3 it only shows 1 carrier without name but cart still has 2 products.
> 
> It allows you to make the order and at the backoffice level it splits the order.
> 
> This is very severe because if you don't allow S1 for a specific country, you still end up the order and the customer makes the payment.

Yes it is! It is a big problem....

**Comment by mikeKstm** [comment]
Still present on **1.7.7.8**? I can't believe this **hasn't been fixed since 2018!!!**

![image](https://user-images.githubusercontent.com/39406664/135057174-ec1a6400-2ae1-4e87-b1d5-0e4ac13b10ba.png)

**Comment by kpodemski** [comment]
Hello @mikeKstm 

This is an open-source project. We're doing our best, but you can help us if you want. You can either fix the problem yourself and contribute via PR or hire a developer to do that. We would highly appreciate this 👍

**Comment by mikeKstm** [comment]
@kpodemski sure I will, thanks for the reply. This is a basic use of carriers, maybe this would be a priority to the comunity before blockreassurance (something aesthetic) in an e-commerce.

Regards

**Comment by msaustral** [comment]
Hi we do have multiples customers with same issue, split orders where one of the order has a deleted address as a shipping address.

Is the a way to disable multiples shipping address on the Core?

Any idea?

**Comment by seederp2p** [comment]
Is this still present??

**Comment by msaustral** [comment]
Hi yes it is, till 1.7.8.1.

This issue is hard to solve as it is random

**Comment by seederp2p** [comment]
> Hi yes it is, till 1.7.8.1.
> 
> This issue is hard to solve as it is random

To me it happens all the time when certain conditions are met:

https://github.com/PrestaShop/PrestaShop/issues/23738

**Comment by seederp2p** [comment]
This bug was open 1241 days ago and we all agree that this is something important to fix.

Why is this so hard to get this fixed?

Thanks

**Comment by matks** [comment]
Hello @seederp2p it seems that this issue is important for you.

If you find that this issue is critical for you, and it's important to have it fixed ASAP, then you can invest into having it done. You can hire a developer to fix it, or if you are a developer yourself, you can try doing it on your own. The most important thing is to share that fix with everyone by submitting a Pull Request -- that's what the open source spirit is all about.

We understand that not having an issue fixed quickly (or in a fixed timeframe) can be frustrating. It does not mean that it won't be fixed, or that we don't care. It doesn't mean that the issue is NOT important either. It just means that other issues have been considered, for whatever reason, more important, so they were (or are being) tackled first. There's an immense amount of work to be done, and tasks take time.

Developer time is limited, so if the amount of issues opened in a certain timeframe exceeds the amount of issues that can be processed using the available developer time, tasks will simply pile up. This is common to ALL projects. You can read https://m.signalvnoise.com/software-has-bugs-this-is-normal/ about this.

So to make it short:
- you want this issue to be fixed soon? then fix it yourself or hire a developer to fix it
- else if you do no want to use your time or resources then wait for someone else to do it

**Comment by msaustral** [comment]
Hi everyone, when Prestashop ask to the community to pay to fix a core issue that is affecting many stores, reported everywhere since 2018, is causing that customer migrate to other platforms and forget about Prestashop.

Please find the FINAL solution to split order:

1- modify class/PaymentModule.php line 320 -+ and add a break before the firs closing } like this:

```
foreach ($package_list as $id_address => $packageByAddress) {
                foreach ($packageByAddress as $id_package => $package) {
                    $orderData = $this->createOrderFromCart(
                        $this->context->cart,
                        $this->context->currency,
                        $package['product_list'],
                        $id_address,
                        $this->context,
                        $reference,
                        $secure_key,
                        $payment_method,
                        $this->name,
                        $dont_touch_amount,
                        $amount_paid,
                        $package_list[$id_address][$id_package]['id_warehouse'],
                        $cart_total_paid,
                        self::DEBUG_MODE,
                        $order_status,
                        $id_order_state,
                        isset($package['id_carrier']) ? $package['id_carrier'] : null
                    );
                    $order = $orderData['order'];
                    $order_list[] = $order;
                    $order_detail_list[] = $orderData['orderDetail'];
                    break; 
                } 
                break; 
            }
```

This will prevent the split order.

2 - Modify class/Cart.php line 322 and 327 and erase the condition AND `id_address_delivery` = ' . (int) $id_address; like this:


```
$sql = 'UPDATE `' . _DB_PREFIX_ . 'cart_product`
SET `id_address_delivery` = ' . (int) $id_address_new . '
WHERE  `id_cart` = ' . (int) $this->id;
Db::getInstance()->execute($sql);

```
```
$sql = 'UPDATE `' . _DB_PREFIX_ . 'customization`
SET `id_address_delivery` = ' . (int) $id_address_new . '
WHERE  `id_cart` = ' . (int) $this->id;
Db::getInstance()->execute($sql);
```


This will force to update the delivery address on the cart.

Hopes it helps!

**Comment by seederp2p** [comment]
> Hello @seederp2p it seems that this issue is important for you.
> 
> If you find that this issue is critical for you, and it's important to have it fixed ASAP, then you can invest into having it done. You can hire a developer to fix it, or if you are a developer yourself, you can try doing it on your own. The most important thing is to share that fix with everyone by submitting a Pull Request -- that's what the open source spirit is all about.
> 
> We understand that not having an issue fixed quickly (or in a fixed timeframe) can be frustrating. It does not mean that it won't be fixed, or that we don't care. It doesn't mean that the issue is NOT important either. It just means that other issues have been considered, for whatever reason, more important, so they were (or are being) tackled first. There's an immense amount of work to be done, and tasks take time.
> 
> Developer time is limited, so if the amount of issues opened in a certain timeframe exceeds the amount of issues that can be processed using the available developer time, tasks will simply pile up. This is common to ALL projects. You can read https://m.signalvnoise.com/software-has-bugs-this-is-normal/ about this.
> 
> So to make it short:
> 
> * you want this issue to be fixed soon? then fix it yourself or hire a developer to fix it
> * else if you do no want to use your time or resources then wait for someone else to do it

Hello, I'm sorry but that's the kind of reply that no one should give.

Saying Prestashop is a free/open-source solution gives an amateur vision of a business model that actually sells modules, support plans, etc.

Try to sell that without a fairly working CMS and you'll what happens...

This issue affects multiple users since 2018!

Nothing personal, but no one should give this kind of reply in this kind of issue.

**Comment by seederp2p** [comment]
> Hi everyone, when Prestashop ask to the community to pay to fix a core issue that is affecting many stores, reported everywhere since 2018, is causing that customer migrate to other platforms and forget about Prestashop.
> 
> Please find the FINAL solution to split order:
> 
> 1- modify class/PaymentModule.php line 320 -+ and add a break before the firs closing } like this:
> 
> ```
> foreach ($package_list as $id_address => $packageByAddress) {
>                 foreach ($packageByAddress as $id_package => $package) {
>                     $orderData = $this->createOrderFromCart(
>                         $this->context->cart,
>                         $this->context->currency,
>                         $package['product_list'],
>                         $id_address,
>                         $this->context,
>                         $reference,
>                         $secure_key,
>                         $payment_method,
>                         $this->name,
>                         $dont_touch_amount,
>                         $amount_paid,
>                         $package_list[$id_address][$id_package]['id_warehouse'],
>                         $cart_total_paid,
>                         self::DEBUG_MODE,
>                         $order_status,
>                         $id_order_state,
>                         isset($package['id_carrier']) ? $package['id_carrier'] : null
>                     );
>                     $order = $orderData['order'];
>                     $order_list[] = $order;
>                     $order_detail_list[] = $orderData['orderDetail'];
>                     break; 
>                 } 
>                 break; 
>             }
> ```
> 
> This will prevent the split order.
> 
> 2 - Modify class/Cart.php line 322 and 327 and erase the condition AND `id_address_delivery` = ' . (int) $id_address; like this:
> 
> ```
> $sql = 'UPDATE `' . _DB_PREFIX_ . 'cart_product`
> SET `id_address_delivery` = ' . (int) $id_address_new . '
> WHERE  `id_cart` = ' . (int) $this->id;
> Db::getInstance()->execute($sql);
> ```
> 
> ```
> $sql = 'UPDATE `' . _DB_PREFIX_ . 'customization`
> SET `id_address_delivery` = ' . (int) $id_address_new . '
> WHERE  `id_cart` = ' . (int) $this->id;
> Db::getInstance()->execute($sql);
> ```
> 
> This will force to update the delivery address on the cart.
> 
> Hopes it helps!

I'll wait for the QA testing. However I thank you for the effort and probably for finding a solution for this! Cheers!!

**Comment by matks** [comment]
> Hello, I'm sorry but that's the kind of reply that no one should give.
> 
> Saying Prestashop is a free/open-source solution gives an amateur vision of a business model that actually sells modules, support plans, etc.

I'm sorry to tell you but what I'm saying is simply facts. You can call it "amateur vision" if you dislike it but it's the truth. This github repository is about the prestashop project, so I'm not trying to sell you anything, modules or support plans.

If nobody from the maintainer team fixed this issue since 2018 it's because it's not important enough for us to do it now. We have other things we need to build and we decided they are more important. So this bug will maybe stay there for a few more years until someone considers it is important enough that it is worth the investment. Maybe it will be you, maybe it will be us, maybe it will be someone else. At some point someone will step up and own this bug because it is worth the money to be fixed.

If nobody thinks this bug is worth the money, then it means this bug is not important enough.

**Comment by msaustral** [comment]
Hi @matks, sorry but this issue doesn't seem to be a problem only to PS, there are no less than 5 different cases reporting the same issue, we've already lost 2 customers because this issue has never been fixed!

Prestashop is used in large store, some do not care if they have a split order, others it is essential not to have it since they lose sales every time a customer complains that their order was not shipped in full.

We know that this is an open source project, but when a company decides to use Prestashop, they want qualified support, usually what they get "is a Prestashop issue reported since 2018 with no solution or comment, nothing to do" from some local companies Usually referred to as "Partners", we have reported it many times, we always try to give customers a solution, even if it is a problem in core of Prestashop.

If you are a client, will you continue working with a solution if the company that creates it does not provide solutions to problems for years?

Every day the market gets smaller, the customer needs a solution, needs someone to respond when things go wrong!

We always defend Prestashop but many times we do not have the necessary tools to do so

**Comment by seederp2p** [comment]
> > Hello, I'm sorry but that's the kind of reply that no one should give.
> > Saying Prestashop is a free/open-source solution gives an amateur vision of a business model that actually sells modules, support plans, etc.
> 
> I'm sorry to tell you but what I'm saying is simply facts. You can call it "amateur vision" if you dislike it but it's the truth. This github repository is about the prestashop project, so I'm not trying to sell you anything, modules or support plans.
> 
> If nobody from the maintainer team fixed this issue since 2018 it's because it's not important enough for us to do it now. We have other things we need to build and we decided they are more important. So this bug will maybe stay there for a few more years until someone considers it is important enough that it is worth the investment. Maybe it will be you, maybe it will be us, maybe it will be someone else. At some point someone will step up and own this bug because it is worth the money to be fixed.
> 
> If nobody thinks this bug is worth the money, then it means this bug is not important enough.

Your reasoning is flawed and is zero "Business oriented". When someone says "if nobody thinks this bug is worth the money"... It tells a lot.

Nobody said you were selling something. but pretending that Prestashop is, at present time, a simple "good will project" with zero profit goals and pretty much "forgetting" that maybe... just maybe, some people that takes the time to report and test stuff here are at the same time "customers" in the sense that they already invested some good amount of money... is pretty bad.

Prestashop makes money to Prestashop SA, to Prestashop Partners, to developers, etc. Not taking care of a product and the end customers is the equivelent to not taking care of a baby (try it :) )

Cheers.

**Comment by seederp2p** [comment]
> Hi @matks, sorry but this issue doesn't seem to be a problem only to PS, there are no less than 5 different cases reporting the same issue, **we've already lost 2 customers** because this issue has never been fixed!
> 
> Prestashop is used in large store, some do not care if they have a split order, others it is essential not to have it since they **lose sales every time a customer complains that their order was not shipped in full**.
> 
> 
> **If you are a client, will you continue working with a solution if the company that creates it does not provide solutions to problems for years?**
> 
> Every day the market gets smaller, **the customer needs a solution, needs someone to respond when things go wrong!**
> 

Just turning into bold the real meaning of making a product "customer ready".

100% agree with this vision.

**Comment by matks** [comment]
Please remember how amazing this is. You benefit from _thousand_ of developers working together to make new versions of PrestaShop all the time for free. Last week, there was [9 new patch releases](https://build.prestashop.com/news/coreweekly-1-2022/) delivered for modules. How much did you pay? Zero. It was all given to you for free. PrestaShop 1.7.8.0, PrestaShop 1.7.8.1, PrestaShop 1.7.8.2... all of this, given to you for free. Isn't this amazing? Can you pause and think about how much money it represent? How much money you would have spent if you had paid someone to build it? You know developer rates are high.

However there's a downside. PrestaShop - the project - is free. You cannot expect qualified support if you pay nothing. Nobody works for free, you understand this from a business point of view.

> We know that this is an open source project, but when a company decides to use Prestashop, they want qualified support, 

So as I said I'm sorry you're wanting something that is not possible. You cannot have PrestaShop which is free and have qualified support. Qualified support comes with paying for it.

If you're willing to have qualified support, the PrestaShop Company provides [support plans](https://www.prestashop.com/support). You pay, they deliver. It's simple business.

That being said I think we covered the full topic. You can disagree with what I say but I simply describe things as they are.

**Comment by seederp2p** [comment]
@matks this leads nowhere.

@msaustral already gave you a perfectly valid reasoning. This issue isn't minor a cosmetic flaw. This is something that leads merchants to potentially lose business.

Ecosystem is the keyword. Care is the keyword. I'll rest my case as I'm pretty sure you understand my point.

Cheers.

**Comment by matks** [comment]
I agree this leads nowhere and we are far away from the point of this web page which is to track the bug.

So let's stop discussing out-of-scope matters. Further discussions on this topic should be about the bug status and the bug progress, nothing else.

**Comment by seederp2p** [comment]
@matks actually @msaustral already provided a suggestion for the solution... so maybe this should be tested and incorporated in the core.

**Comment by seederp2p** [comment]
@matks can you please tell us what does it take to get the suggestion from @msaustral tested and introduced into the core?

Thanks

**Comment by matks** [comment]
How to contribute is documented on [this page](https://devdocs.prestashop.com/8/contribute/contribute-pull-requests/).

**Comment by seederp2p** [comment]
> How to contribute is documented on [this page](https://devdocs.prestashop.com/8/contribute/contribute-pull-requests/).

@msaustral are you willing to do that so maybe we can finally have a fix for this?

Thanks!

**Comment by Caronte-Studio** [comment]
Has this been fixed already?

I tried @msaustral solution:

> > 1- modify class/PaymentModule.php line 320 -+ and add a break before the firs closing } like this:
> > ```
> > foreach ($package_list as $id_address => $packageByAddress) {
> >                 foreach ($packageByAddress as $id_package => $package) {
> >                     $orderData = $this->createOrderFromCart(
> >                         $this->context->cart,
> >                         $this->context->currency,
> >                         $package['product_list'],
> >                         $id_address,
> >                         $this->context,
> >                         $reference,
> >                         $secure_key,
> >                         $payment_method,
> >                         $this->name,
> >                         $dont_touch_amount,
> >                         $amount_paid,
> >                         $package_list[$id_address][$id_package]['id_warehouse'],
> >                         $cart_total_paid,
> >                         self::DEBUG_MODE,
> >                         $order_status,
> >                         $id_order_state,
> >                         isset($package['id_carrier']) ? $package['id_carrier'] : null
> >                     );
> >                     $order = $orderData['order'];
> >                     $order_list[] = $order;
> >                     $order_detail_list[] = $orderData['orderDetail'];
> >                     break; 
> >                 } 
> >                 break; 
> >             }
> >   ```
> > This will prevent the split order.
> > 2 - Modify class/Cart.php line 322 and 327 and erase the condition AND `id_address_delivery` = ' . (int) $id_address; like this:
> > ```
> > $sql = 'UPDATE `' . _DB_PREFIX_ . 'cart_product`
> > SET `id_address_delivery` = ' . (int) $id_address_new . '
> > WHERE  `id_cart` = ' . (int) $this->id;
> > Db::getInstance()->execute($sql);
> > $sql = 'UPDATE `' . _DB_PREFIX_ . 'customization`
> > SET `id_address_delivery` = ' . (int) $id_address_new . '
> > WHERE  `id_cart` = ' . (int) $this->id;
> > Db::getInstance()->execute($sql);
> > ```

But unfortunately, I couldn't get it to work. This is my PaymentModule.php code:
```
foreach ($package_list as $id_address => $packageByAddress) {
                foreach ($packageByAddress as $id_package => $package) {
                    /** @var Order $order */
                    $order = new Order();
                    $order->product_list = $package['product_list'];

                    if (Configuration::get('PS_TAX_ADDRESS_TYPE') == 'id_address_delivery') {
                        $address = new Address((int) $id_address);
                        $this->context->country = new Country((int) $address->id_country, (int) $this->context->cart->id_lang);
                        if (!$this->context->country->active) {
                            throw new PrestaShopException('The delivery address country is not active.');
                        }
                    }

                    $carrier = null;
                    if (!$this->context->cart->isVirtualCart() && isset($package['id_carrier'])) {
                        $carrier = new Carrier((int) $package['id_carrier'], (int) $this->context->cart->id_lang);
                        $order->id_carrier = (int) $carrier->id;
                        $id_carrier = (int) $carrier->id;
                    } else {
                        $order->id_carrier = 0;
                        $id_carrier = 0;
                    }

                    $order->id_customer = (int) $this->context->cart->id_customer;
                    $order->id_address_invoice = (int) $this->context->cart->id_address_invoice;
                    $order->id_address_delivery = (int) $id_address;
                    $order->id_currency = $this->context->currency->id;
                    $order->id_lang = (int) $this->context->cart->id_lang;
                    $order->id_cart = (int) $this->context->cart->id;
                    $order->reference = $reference;
                    $order->id_shop = (int) $this->context->shop->id;
                    $order->id_shop_group = (int) $this->context->shop->id_shop_group;

                    $order->secure_key = ($secure_key ? pSQL($secure_key) : pSQL($this->context->customer->secure_key));
                    $order->payment = $payment_method;
                    if (isset($this->name)) {
                        $order->module = $this->name;
                    }
                    $order->recyclable = $this->context->cart->recyclable;
                    $order->gift = (int) $this->context->cart->gift;
                    $order->gift_message = $this->context->cart->gift_message;
                    $order->mobile_theme = $this->context->cart->mobile_theme;
                    $order->conversion_rate = $this->context->currency->conversion_rate;
                    $amount_paid = !$dont_touch_amount ? Tools::ps_round((float) $amount_paid, 2) : $amount_paid;
                    $order->total_paid_real = 0;

                    $order->total_products = (float) $this->context->cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
                    $order->total_products_wt = (float) $this->context->cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $order->product_list, $id_carrier);
                    $order->total_discounts_tax_excl = (float) abs($this->context->cart->getOrderTotal(false, Cart::ONLY_DISCOUNTS, $order->product_list, $id_carrier));
                    $order->total_discounts_tax_incl = (float) abs($this->context->cart->getOrderTotal(true, Cart::ONLY_DISCOUNTS, $order->product_list, $id_carrier));
                    $order->total_discounts = $order->total_discounts_tax_incl;

                    $order->total_shipping_tax_excl = (float) $this->context->cart->getPackageShippingCost((int) $id_carrier, false, null, $order->product_list);
                    $order->total_shipping_tax_incl = (float) $this->context->cart->getPackageShippingCost((int) $id_carrier, true, null, $order->product_list);
                    $order->total_shipping = $order->total_shipping_tax_incl;

                    if (!is_null($carrier) && Validate::isLoadedObject($carrier)) {
                        $order->carrier_tax_rate = $carrier->getTaxesRate(new Address((int) $this->context->cart->{Configuration::get('PS_TAX_ADDRESS_TYPE')}));
                    }

                    $order->total_wrapping_tax_excl = (float) abs($this->context->cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $order->product_list, $id_carrier));
                    $order->total_wrapping_tax_incl = (float) abs($this->context->cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $order->product_list, $id_carrier));
                    $order->total_wrapping = $order->total_wrapping_tax_incl;

                    $order->total_paid_tax_excl = (float) Tools::ps_round((float) $this->context->cart->getOrderTotal(false, Cart::BOTH, $order->product_list, $id_carrier), _PS_PRICE_COMPUTE_PRECISION_);
                    $order->total_paid_tax_incl = (float) Tools::ps_round((float) $this->context->cart->getOrderTotal(true, Cart::BOTH, $order->product_list, $id_carrier), _PS_PRICE_COMPUTE_PRECISION_);
                    $order->total_paid = $order->total_paid_tax_incl;
                    $order->round_mode = Configuration::get('PS_PRICE_ROUND_MODE');
                    $order->round_type = Configuration::get('PS_ROUND_TYPE');

                    $order->invoice_date = '0000-00-00 00:00:00';
                    $order->delivery_date = '0000-00-00 00:00:00';

                    if (self::DEBUG_MODE) {
                        PrestaShopLogger::addLog('PaymentModule::validateOrder - Order is about to be added', 1, null, 'Cart', (int) $id_cart, true);
                    }

                    // Creating order
                    $result = $order->add();

                    if (!$result) {
                        PrestaShopLogger::addLog('PaymentModule::validateOrder - Order cannot be created', 3, null, 'Cart', (int) $id_cart, true);
                        throw new PrestaShopException('Can\'t save Order');
                    }

                    // Amount paid by customer is not the right one -> Status = payment error
                    // We don't use the following condition to avoid the float precision issues : http://www.php.net/manual/en/language.types.float.php
                    // if ($order->total_paid != $order->total_paid_real)
                    // We use number_format in order to compare two string
                    if ($order_status->logable && number_format($cart_total_paid, _PS_PRICE_COMPUTE_PRECISION_) != number_format($amount_paid, _PS_PRICE_COMPUTE_PRECISION_)) {
                        $id_order_state = Configuration::get('PS_OS_ERROR');
                    }

                    $order_list[] = $order;

                    if (self::DEBUG_MODE) {
                        PrestaShopLogger::addLog('PaymentModule::validateOrder - OrderDetail is about to be added', 1, null, 'Cart', (int) $id_cart, true);
                    }

                    // Insert new Order detail list using cart for the current order
                    $order_detail = new OrderDetail(null, null, $this->context);
                    $order_detail->createList($order, $this->context->cart, $id_order_state, $order->product_list, 0, true, $package_list[$id_address][$id_package]['id_warehouse']);
                    $order_detail_list[] = $order_detail;

                    if (self::DEBUG_MODE) {
                        PrestaShopLogger::addLog('PaymentModule::validateOrder - OrderCarrier is about to be added', 1, null, 'Cart', (int) $id_cart, true);
                    }

                    // Adding an entry in order_carrier table
                    if (!is_null($carrier)) {
                        
                        $order_carrier = new OrderCarrier();
                        $order_carrier->id_order = (int) $order->id;
                        $order_carrier->id_carrier = (int) $id_carrier;
                        $order_carrier->weight = (float) $order->getTotalWeight();
                        $order_carrier->shipping_cost_tax_excl = (float) $order->total_shipping_tax_excl;
                        $order_carrier->shipping_cost_tax_incl = (float) $order->total_shipping_tax_incl;
                        $order_carrier->add();
                    }
                
                }
            }
```


Any ideas on how to fix it? 
Thanks in advance.

**Comment by msaustral** [comment]
Hi, what PS version do you have?

**Comment by Caronte-Studio** [comment]
Hi,
The version of my PrestaShop is 1.7.5.2.

**Comment by msaustral** [comment]
Hi, that is not the original class/PaymentModule.php did you migrate from PS 1.6? did you override the class?, try to download PS ver 1.7.5.2 from prestashop.com and replace the class with the original.

**Comment by Caronte-Studio** [comment]
Yes I have migrated from version 1.6, thank you, I will try.

**Comment by msaustral** [comment]
Hi, substitute the class with the original package

**Comment by Caronte-Studio** [comment]
I have downloaded the 1.7.5.2 version file, but the loop is the same, I have tried adding the break, but it still doesn't work.

**Comment by msaustral** [comment]
We migrate from 1.6, not all file are change to the new version, make a fresh instillation, migrate the data and apply the fix

**Comment by Bavarianspirit** [comment]
there is a issue assigned for Milestone 9.0.0.
Display multiple carriers properly in checkout #36032 
These Code Changes fixed my issue on Prestashop 8.1.5

**Comment by jouino16** [comment]
I would like to notify you that I have developed a solution to the problem cited by @prestonBot. I invite you to take a look at the module on PrestaShop addons [https://addons.prestashop.com/en/shipping-carriers/93768-multi-shipping-multiple-carriers-order.html](https://addons.prestashop.com/en/shipping-carriers/93768-multi-shipping-multiple-carriers-order.html) or here [https://smartunicorn.fr/produit/module-prestashop-commande-avec-transporteurs-livraisons-multiples/](https://smartunicorn.fr/produit/module-prestashop-commande-avec-transporteurs-livraisons-multiples/) (**more readable FAQ part here to read**).

The order will be split according to the number of different carriers involved, while keeping a display taking into account the different packages throughout the purchasing process (order page and order confirmation page). I invite you to [watch the demo video](https://www.youtube.com/watch?v=5QQMzO3-EdI&ab_channel=SmartUnicorn) and read the documentation [Multi Shipping - Multiple Carriers Order_documentation.pdf](https://github.com/user-attachments/files/19738015/Multi.Shipping.-.Multiple.Carriers.Order_documentation.pdf).

I hope this will unblock the situation for some of you.

**Comment by seederp2p** [comment]
> I would like to notify you that I have developed a solution to the problem cited by [@prestonBot](https://github.com/prestonBot). I invite you to take a look at the module on PrestaShop addons https://addons.prestashop.com/en/shipping-carriers/93768-multi-shipping-multiple-carriers-order.html or here https://smartunicorn.fr/produit/module-prestashop-commande-avec-transporteurs-livraisons-multiples/ (**more readable FAQ part here to read**).
> 
> The order will be split according to the number of different carriers involved, while keeping a display taking into account the different packages throughout the purchasing process (order page and order confirmation page). I invite you to watch the demo video and read the documentation.
> 
> I hope this will unblock the situation for some of you.

Maybe you could contribute to the PrestaShop core instead?

I don’t think promoting a €252 commercial module adds much value to the project...

**Comment by boo-code** [comment]
The status of this one is recorded in the code, and since the thread keeps asking, here it is with the current numbers from 9.2.x.

`OrderConfirmationController::init()` carries this above the lookup:

> Because of order splitting scenarios, we must get the data by id_cart parameter (not id_order), so we can display all orders made from this cart.
>
> It's not implemented yet, however, and probably won't be, because we are switching to a new logic of multiple shipments per order, which doesn't require splitting orders anymore.

It then calls `Order::getIdByCartId()`, which returns one id, and builds the page from that single order. That is why only one package's items appear.

Where the replacement stands, measured rather than assumed:

- `install-dev/data/xml/feature_flag.xml` declares `improved_shipment` with `state="0"` and `stability="beta"`, so it is off on a default install.
- With it off, `PaymentModule::validateOrder()` takes the branch at line 338 and calls `createOrderFromCart()` once per package, producing several orders. This is what every default shop does today, so the symptom here is live.
- With it on, the branch at line 364 groups the packages by carrier and calls `createOrderFromCart()` once, outside any loop. One order is created, so the confirmation page has nothing to split and the symptom does not arise.
- The work is active: #42418, #42411, #42092, #42414 and #39631 all moved in the last few days.

So this is not unowned, it is superseded. A patch making the confirmation page render every order of a split cart would add code to the path being removed, which is why I am not opening one. If a maintainer would rather have that display fixed for the flag-off path while the migration finishes, say so and I will write it.
